docs/Guides/Testing.md

```diff
@@ -243,33 +243,72 @@ after initializing routes and plugins with `fastify.ready()`.
 
 Uses **app.js** from the previous example.
 
-**test-listen.js** (testing with
-[`Request`](https://www.npmjs.com/package/request))
+**test-listen.js** (testing with [`undici`](https://www.npmjs.com/package/undici))
 ```js
 const tap = require('tap')
-const request = require('request')
+const { Client } = require('undici')
 const buildFastify = require('./app')
 
-tap.test('GET `/` route', t => {
-  t.plan(5)
+tap.test('should work with undici', async t => {
+  t.plan(2)
 
   const fastify = buildFastify()
 
-  t.teardown(() => fastify.close())
+  await fastify.listen()
 
-  fastify.listen({ port: 0 }, (err) => {
-    t.error(err)
+   const client = new Client(
+    'http://localhost:' + fastify.server.address().port, {
+      keepAliveTimeout: 10,
+      keepAliveMaxTimeout: 10
+    } 
+  )
 
-    request({
-      method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port
-    }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.equal(response.headers['content-type'], 'application/json; charset=utf-8')
-      t.same(JSON.parse(body), { hello: 'world' })
-    })
+  t.teardown(() => {
+    fastify.close()
+    client.close()
   })
+
+  const response = await client.request({ method: 'GET', path: '/' })
+
+  t.equal(await response.body.text(), '{"hello":"world"}')
+  t.equal(response.statusCode, 200)
+})
+```
+
+#### Another Example:
+
+This example is a copy of the previous one, this time using `fetch` 
+to make the request. 
+
+> **_NOTE:_** `fetch` is available in Node 18
+
+Uses **app.js** from the previous example.
+
+**test-listen.js** (testing with
+[`fetch`](https://nodejs.org/docs/latest-v18.x/api/globals.html#fetch))
```

Comment:
I don't know why this suggestion didn't get added in my last review. It was supposed to be. GitHub just decided to ignore it completely 😕 

```suggestion

Alternatively, starting with Node.js 18, [`fetch`](https://nodejs.org/docs/latest-v18.x/api/globals.html#fetch) may be used without requiring any extra dependencies:

**test-listen.js**
```